server/server_test.go

```diff
@@ -2271,6 +2273,7 @@ func TestPoliciesPathSlashes(t *testing.T) {
 
 func TestQueryPostBasic(t *testing.T) {
 	f := newFixture(t)
+	metrics.ResetGlobalMetricsRegistry()
```

Comment:
Would it make sense to just add the reset to `newFixture()`? Or maybe even on the server `initRouter()` or `New()`?

It seems like the lifetime of the metrics registry should also be tied with the server, right? I'm not sure we really support running >1 server at a time anyway. (@tsandall and others correct me if I'm wrong on that one 😃 )

Reply:
I added Reset() to the newFixture() but not all tests use it or (like in this case) create another server instance after that. Putting Reset() in the server itself would be incorrect as the registry might already have collectors registered from other parts of the code. The whole idea was to move the registry out of the server so the server shouldn't control it. Reset() is needed only for the tests